src/test/java/org/opensearch/knn/index/KNNSettingsTests.java

```diff
@@ -196,11 +196,11 @@ private static Settings.Builder baseSettings() {
             .put(dataNode());
     }
 
-    private void assertWarnings() {
+    private void allowedWarnings() {
         // set warning for deprecation of index.store.hybrid.mmap.extensions as expected temporarily, need to work on proper strategy of
         // switching to new setting in core
         // no-jdk distributions expected warning is a workaround for running tests locally
-        assertWarnings(
+        allowedWarnings(
```

Comment:
I don't think this is a right approach. Before, we wanted to make sure that the warning message is generated so that customer get to know that it will be deprecated. Now, we only allow this warning message but nothing else. Then, even if there is no warning message, it will pass.

Reply:
Thanks the context sharing @heemin32, the PR is raised from https://github.com/opensearch-project/k-NN/issues/1392#issuecomment-1919622685, do you have any other suggestion to mitigate the flaky issue?

Reply:
Best way is to find out why warning message is not printed sometimes. If it takes time, we can disabled the check and create a separate issue to follow up.

Reply:
@heemin32 Sure, will try to reproduce the issue from our side, we might need more details on the flaky issue from @salyh. https://github.com/opensearch-project/k-NN/issues/1392#issuecomment-1921775181